### ISSUE TITLE
DuplicateSubnetsInSameZone while trying to add subnets

### ISSUE THREAD

**Issue by wavemoran** [terraform-aws-modules/terraform-aws-vpc#583]
I'm getting a `DuplicateSubnetsInSameZone` error while trying to add some additional subnets to my VPC, since the `ecr_api_endpoint_subnet_ids` variable by default tries to reference all private subnets.

How do I get `ecr_api_endpoint_subnet_ids` to only reference private subnets 0-2 from the output list?

**Comment by bryantbiggs** [comment]
hi @wavemoran - can you share your configuration? its hard to follow exactly what issue you are facing

**Comment by wavemoran** [comment]
Sure, I've been using the module for a while with only three private subnets defined. I have a few VPC endpoints set up as well:

```hcl
  enable_ecr_api_endpoint              = true
  ecr_api_endpoint_private_dns_enabled = true
```

I've gone in to expand the number of private subnets from three to six:

```hcl
private_subnets = ["10.13.1.0/24", "10.13.2.0/24", "10.13.3.0/24", "10.13.4.0/24", "10.13.5.0/24", "10.13.6.0/24"]
```

When I try to apply this new configuration, I get the following error:

```
Error: Error updating VPC Endpoint: DuplicateSubnetsInSameZone: Found another VPC endpoint subnet in the availability zone of subnet-004915fc13af2e4dc. VPC endpoint subnets should be in different availability zones supported by the VPC endpoint service.
	status code: 400, request id: 055f3ad5-69c0-4df6-973d-d99dfbd78bbe
```

In the module, it looks as though the default for endpoint subnet IDs is to reference all private subnets. This isn't allowed by AWS as only a single subnet in an AZ can be enabled for service endpoints.

So my question is, am I missing something in my configuration or is there a way to define the variable input, eg `ecr_api_endpoint_subnet_ids` to only reference a subset of the `private_subnets` variable?

**Comment by bryantbiggs** [comment]
ah, ok now I see what you are running into. so yes - you basically added a 2nd set of CIDR blocks for the private subnets which means in each AZ you have two private subnets and therefore the error is showing up. if you can, the easiest fix would be to increase your existing 3 subnets to be something like /23 or /22 though that can be destructive. otherwise, I'll have to think about how we can handle this better

**Comment by sksumit1** [comment]
I did a workaround to not use the module to do the endpoint setup.
Can we expect a fix for the issue anytime soon?

**Comment by bryantbiggs** [comment]
if you upgrade to v3.0.0 you can utilize the new configuration for a more flexible workflow with VPC endpoints
